src/utils/compute.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  DDO,
+  Ocean,
+  ServiceComputePrivacy,
+  publisherTrustedAlgorithm as PublisherTrustedAlgorithm
+} from '@oceanprotocol/lib'
+import { ComputePrivacyForm } from '../models/FormEditComputeDataset'
+
+export async function createTrustedAlgorithmList(
+  selectedAlgorithms: string[], // list of DIDs
+  ocean: Ocean
+): Promise<PublisherTrustedAlgorithm[]> {
+  const trustedAlgorithms = []
+
+  for (const selectedAlgorithm of selectedAlgorithms) {
+    const trustedAlgorithm = await ocean.compute.createPublisherTrustedAlgorithmfromDID(
+      selectedAlgorithm
+    )
+    trustedAlgorithms.push(trustedAlgorithm)
+  }
+  return trustedAlgorithms
+}
+
+export async function transformComputeFormToServiceComputePrivacy(
+  values: ComputePrivacyForm,
+  ddo: DDO,
+  ocean: Ocean
+): Promise<ServiceComputePrivacy> {
+  // pass through allowRawAlgorithm & allowNetworkAccess from DDO
```

Comment:
I'm tending towards ditching that and setting both, `allowRawAlgorithm` & `allowNetworkAccess` further down in the final `privacy` return to `false` by default. While those will be the defaults for every asset published through our market, we would then need to make sure to switch them depending on if `allowAllPublishedAlgorithms` is set or not. Since we do not expose these 2 options right now most simple way should be to just set them always to `false`